src/main/java/org/jabref/logic/importer/fetcher/Medra.java

```diff
@@ -0,0 +1,173 @@
+                                                                                       .map(array -> array.optInt(0))
+                                                                                       .map(year -> Integer.toString(year)).orElse(""))
+                                                                    .withField(StandardField.DOI, item.getString("DOI"))
+                                                                    .withField(StandardField.PAGES, item.optString("page"))
+                                                                    .withField(StandardField.ISSN, item.optString("ISSN"))
+                                                                    .withField(StandardField.JOURNAL, item.optString("container-title"))
+                                                                    .withField(StandardField.PUBLISHER, item.optString("publisher"))
+                                                                    .withField(StandardField.URL, item.optString("URL"))
+                                                                    .withField(StandardField.VOLUME, item.optString("volume"));
+
+        } catch (JSONException exception) {
+            throw new ParseException("mEdRA API JSON format has changed", exception);
+        }
+    }
+
+    private EntryType convertType(String type) {
+        switch (type) {
+            case "article-journal":
+                return StandardEntryType.Article;
+            default:
+                return StandardEntryType.Misc;
+        }
+    }
+
+    private String toAuthors(JSONArray authors) {
+        if (authors == null) {
+            return "";
+        }
+
+        // input: list of {"literal":"A."}
+        AuthorList authorsParsed = new AuthorList();
+        String name = "";
+
+        for (int i = 0; i < authors.length(); i++) {
+            JSONObject author = authors.getJSONObject(i);
+            name = author.optString("literal", "") + " " + author.optString("family", "") + " " + author.optString("given", "");
+
+            authorsParsed.addAuthor(
+                                    name,
+                                    "",
+                                    "",
+                                    "",
+                                    "");
+
+        }
+        return authorsParsed.getAsFirstLastNamesWithAnd();
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        if (StringUtil.isBlank(identifier)) {
+            return Optional.empty();
+        }
+
+        try (InputStream stream = getUrlDownload(identifier).asInputStream();
+             PushbackInputStream pushbackInputStream = new PushbackInputStream(stream)) {
+
+            List<BibEntry> fetchedEntries = new ArrayList<>();
+
+            // check if there is anything to read
+            int readByte;
+            readByte = pushbackInputStream.read();
```

Comment:
Could you please comment why this was done?

I would bet that `getParser().parseEntries(pushbackInputStream)` works for an empty input stream (and does not throw any exceptions).

Oh, I lost my bet:

![grafik](https://user-images.githubusercontent.com/1366654/86732804-fd5e4b80-c030-11ea-9b51-a72a57f4818d.png)

Maybe, change line 28 to return an empty JSonObject and comment the JavaDoc accordingly. --> An empty input stream results in an empty JsonObject and not into an error. With that change, the whole "hack" with with PushbackInputStream should not be required anymore.

Maybe, there is a test case missing demonstrating that functionality.

The UrlDownload returns an empty input stream to avoid such code:

![grafik](https://user-images.githubusercontent.com/1366654/86731890-3c3fd180-c030-11ea-9acb-96baa723741d.png)

Reply:
Sure, if I modify JsonReader this would fix empty stream exception for JSON parsing. In this way i can remove the hack. I could log 'Empty input!' and return the empty JsonObject.

Exactly, in CrossRefTest there wasn't a test case for this, that's why I've inserted one in the last commit.

Reply:
Fixed in https://github.com/mind000/jabref/pull/1.